pkg/manager/manager.go

```diff
@@ -560,15 +561,52 @@ func (s *FraktiManager) Status(ctx context.Context, req *kubeapi.StatusRequest)
 func (s *FraktiManager) ListImages(ctx context.Context, req *kubeapi.ListImagesRequest) (*kubeapi.ListImagesResponse, error) {
 	glog.V(3).Infof("ListImages with request %s", req.String())
 
-	// NOTE: we only show image list of hyper runtime and assume alternative runtime images are the same
-	images, err := s.imageService.ListImages(req.GetFilter())
-	if err != nil {
-		glog.Errorf("ListImages from hyper image service failed: %v", err)
-		return nil, err
+	var (
+		lock       sync.Mutex
+		firstError error
+	)
+	catchError := func(err error) {
+		lock.Lock()
+		defer lock.Unlock()
+		if firstError == nil {
+			firstError = err
+		}
+	}
+
+	imageServiceList := []runtime.ImageService{s.imageService, s.alternativeRuntimeService}
+	serviceNameList := []string{"hyper image service", "alternative image service"}
+	imageMapList := make([]map[string]*kubeapi.Image, 2)
+
+	listImageFunc := func(i int) {
+		images, err := imageServiceList[i].ListImages(req.GetFilter())
+		if err != nil {
+			catchError(fmt.Errorf("ListImage from %s failed: %v", serviceNameList[i], err))
```

Comment:
nit: a more readable way to handle errors https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/core/generic_scheduler.go#L182